Generate.py

```diff
@@ -256,18 +333,225 @@ def get_choice_legacy(option, root, value=None) -> Any:
     raise RuntimeError(f"All options specified in \"{option}\" are weighted as zero.")
 
 
-def get_choice(option, root, value=None) -> Any:
-    if option not in root:
-        return value
-    if type(root[option]) is list:
-        return random.choices(root[option])[0]
-    if type(root[option]) is not dict:
-        return root[option]
-    if not root[option]:
-        return value
-    if any(root[option].values()):
-        return random.choices(list(root[option].keys()), weights=list(map(int, root[option].values())))[0]
-    raise RuntimeError(f"All options specified in \"{option}\" are weighted as zero.")
+BinOp = Literal["+", "-", "*", "/"]
+
+
+_bops: Mapping[BinOp, Callable[[Union[float, int], Union[float,int]], Union[float, int]]] = {
+    "+": lambda a, b: a + b,
+    "-": lambda a, b: a - b,
+    "*": lambda a, b: a * b,
+    "/": lambda a, b: a / b,
+}
+""" binary operators """
+
+
+def _is_bin_op(x: object) -> TypeGuard[BinOp]:
+    return x in _bops
+
+
+_precedence: Mapping[Union[BinOp, Literal["("]], int] = {
+    "(": 0,
+    "+": 1,
+    "-": 1,
+    "*": 2,
+    "/": 2,
+}
+
+
+@dataclass(frozen=True)
+class Token:
+    val: Union[str, float]
+
+
+@dataclass(frozen=True)
+class NumberToken(Token):
+    val: float
+
+
+@dataclass(frozen=True)
+class NameToken(Token):
+    val: str
+
+
+@dataclass(frozen=True)
+class BinOpToken(Token):
+    val: BinOp
+
+
+@dataclass(frozen=True)
+class OpenParenToken(Token):
+    val: Literal["("]
+
+
+@dataclass(frozen=True)
+class CloseParenToken(Token):
+    val: Literal[")"]
+
+
+AllTokens = Union[NumberToken, NameToken, BinOpToken, OpenParenToken, CloseParenToken]
+PostfixTokens = Union[NameToken, NumberToken, BinOpToken]
+
+
+def parse_tokens(s: str) -> Sequence[AllTokens]:
+    tokens: List[AllTokens] = []
+    state: Literal["start", "end"] = "start"  # sub-expression
+    paren_count = 0
+    i = 0
+    while i < len(s):
+        if s[i].isspace():
+            i += 1
+            continue
+        if state == "start":
+            if s[i] == "(":
+                tokens.append(OpenParenToken("("))
+                paren_count += 1
+                i += 1
+            elif s[i].isdigit() or s[i] == "." or s[i] == "-":
+                j = i + 1
+                while j < len(s) and (s[j].isdigit() or s[j] == "."):
+                    j += 1
+                val_str = s[i:j]
+                if len(val_str.split(".")) > 1:
+                    val = float(val_str)
+                else:
+                    val = int(val_str)
```

Comment:
```suggestion
                val = float(val_str)
```
can simplify this since it will be turned into `int` at the end if needed